### PR TITLE
Test Cases for Openapi Summary, Deprecated, and External Docs; Fix External Documents

### PR DESCRIPTION
**Pull request by ChihweiLHBird**
1. Add test cases for openapi `summary`, `deprecated`, and `externalDocs`.
2. Fix `@openapi.document()` API's issue with an `ExternalDocumentation` object.

Issue found and fixed in this PR: #37.

### CODE REVIEW

**Comment by ChihweiLHBird** [comment]
The fix to the issue #37 is included here.

File: sanic_ext/extensions/openapi/openapi.py [hunk 1]
```diff
@@ -98,8 +98,14 @@ def inner(func):
 
 
 def document(url: Union[str, ExternalDocumentation], description: str = None):
+    if isinstance(url, ExternalDocumentation):
+        _url = url.fields["url"]
+        description = url.fields["description"]
+    elif isinstance(url, str):
+        _url = url
```
Comment (ahopkins) [inline sanic_ext/extensions/openapi/openapi.py:105]: There's no else which could lead to variable not being declared.

File: sanic_ext/extensions/openapi/types.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def __init__(self, **kwargs):
     def fields(self):
         return self._fields
 
-    def guard(self, fields):
+    def guard(self, fields: dict):
```
Comment (ahopkins) [inline sanic_ext/extensions/openapi/types.py:28]: `dict` is not an acceptable annotation.

**Review by ahopkins — COMMENTED** [review]
Nice 🤘

File: sanic_ext/extensions/openapi/openapi.py (hunk 1 above)
Comment (ChihweiLHBird) [reply sanic_ext/extensions/openapi/openapi.py]: Thank you. The logic was improved.

File: sanic_ext/extensions/openapi/types.py (hunk 2 above)
Comment (ChihweiLHBird) [reply sanic_ext/extensions/openapi/types.py]: Removed. May I know why? And what would be a suitable type hint for it?

**Review by ahopkins — APPROVED** [review]

File: sanic_ext/extensions/openapi/types.py (hunk 2 above)
Comment (ahopkins) [reply sanic_ext/extensions/openapi/types.py]: 1. Python 3.8 should have `Dict` and not `dict`
2. It should explain `what` it is: `Dict[str, Any]` being a broad generalization

**Pull request merged**